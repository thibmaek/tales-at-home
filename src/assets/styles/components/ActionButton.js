import { StyleSheet } from 'react-native';
import { globalStyle,
  white, accentRed, accentGreen, buttonShadowColor,
} from 'src/assets/styles/vars';

export default StyleSheet.create({
  buttonContainer: {
    position: `absolute`, bottom: 20, right: 30,
    zIndex: 4,
  },
  circleButton: {
    width: 35, height: 35,
    backgroundColor: accentRed,
    borderWidth: 0, borderRadius: 50,
    shadowColor: buttonShadowColor, shadowOpacity: .3,
    shadowOffset: { width: 0, height: 3 }, shadowRadius: 3,
  },
  circleButtonText: {
    ...globalStyle,
    fontWeight: `bold`,
    color: white,
  },
  button: {
    height: 35,
    paddingLeft: 20, paddingRight: 20,
    backgroundColor: accentGreen,
    borderWidth: 0, borderRadius: 50,
    shadowColor: buttonShadowColor, shadowOpacity: .3,
    shadowOffset: { width: 0, height: 3 }, shadowRadius: 3,
  },
  buttonText: {
    ...globalStyle,
    fontWeight: `bold`,
    color: white,
  },
});
