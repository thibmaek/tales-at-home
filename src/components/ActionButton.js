import React, { Component, PropTypes } from 'react';
import { View, Text, Image } from 'react-native';
import Button from 'apsl-react-native-button';

import s from 'src/assets/styles/components/ActionButton';

import closeIcon from 'src/assets/img/icons/cross.png';

export default class ActionButton extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = { displayChildren: true, displayOptions: false };
  }

  _toggleChildren() {
    this.setState({
      displayChildren: !this.state.displayChildren,
    });
  }

  _toggleOptions() {
    this.setState({
      displayOptions: !this.state.displayOptions,
    });
  }

  render() {
    return (
      <View style={s.actionContainer}>
        <View style={s.optionContainer}>
          <View style={s.headerBar}>
            <Text style={s.title}>Nieuwe sessie starten</Text>
            <Image style={s.closeButton} source={closeIcon} />
          </View>
          {this.state.displayChildren ? this.props.children : null}
        </View>
        {this.props.type === `add`
          ? <Button
              style={s.circleButton} textStyle={s.circleButtonText}
              onPress={() => this._toggleChildren()}
            >
              +
            </Button>
          : <Button style={s.button} textStyle={s.buttonText}>{ this.props.type }</Button>
        }
      </View>
    );
  }

  static propTypes = {
    children: PropTypes.array,
    type: PropTypes.string,
  }
}
