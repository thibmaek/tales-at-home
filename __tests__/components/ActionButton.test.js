import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ActionButton from 'src/components/ActionButton';

jest.mock(`apsl-react-native-button`, () => `Button`);

describe(`Component: <ActionButton>`, () => {

  it(`should render without children passed`, () => {
    const tree = renderer.create(<ActionButton />).toJSON();
    expect(tree).toMatchSnapshot(`ActionButton`);
  });

  it(`should render wrapping, without children passed`, () => {
    const tree = renderer.create(<ActionButton></ActionButton>).toJSON();
    expect(tree).toMatchSnapshot(`ActionButton: Wrapping`);
  });

});
