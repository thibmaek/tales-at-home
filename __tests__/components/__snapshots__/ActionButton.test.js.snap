exports[`ActionButton 1`] = `
<View
  style={
    Object {
      "bottom": 20,
      "position": "absolute",
      "right": 30,
      "zIndex": 4,
    }
  }>
  <Button
    style={
      Object {
        "backgroundColor": "rgb(35, 177, 41)",
        "borderRadius": 50,
        "borderWidth": 0,
        "height": 35,
        "paddingLeft": 20,
        "paddingRight": 20,
        "shadowColor": "rgb(74, 74, 74)",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    textStyle={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontFamily": "Bariol",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    } />
</View>
`;

exports[`ActionButton: Wrapping 1`] = `
<View
  style={
    Object {
      "bottom": 20,
      "position": "absolute",
      "right": 30,
      "zIndex": 4,
    }
  }>
  <Button
    style={
      Object {
        "backgroundColor": "rgb(35, 177, 41)",
        "borderRadius": 50,
        "borderWidth": 0,
        "height": 35,
        "paddingLeft": 20,
        "paddingRight": 20,
        "shadowColor": "rgb(74, 74, 74)",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    textStyle={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontFamily": "Bariol",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    } />
</View>
`;
